### ISSUE TITLE
Feature request: Modi supported

### ISSUE THREAD

**Issue by skank01** [matijse/eufy-ha-mqtt-bridge#19]
Again no rush, but if we could change the modi "home, away , ..."  too within HA... it would rock.
No idea if that will be possible

**Comment by timdonovanuk** [comment]
Also things like "silence chime" etc. 

I'm unsure if stuff like this can even be supported though? Not sure how this bridge works, if it simply captures events or can do two way communication.

**Comment by mikedrawback** [comment]
There are examples in the eufy node client of changing the mode, so it might be possible eventually

[eufy node client examples - p2p change mode](https://github.com/JanLoebel/eufy-node-client-examples/tree/master/examples/p2p-local)

**Comment by r77s** [comment]
Yes. This would be a nice feature, because current the eufy app con not mix geofencing and time based mode. With this feature, this function can be automated in HA.

**Comment by matijse** [comment]
Yes this one is next on my todo list :) It's a bit more tricky, because it looks like it is only possible to send this command over the local network directly to the Home Base. I don't actually own any Eufy camera's, I was just developing this for friends and up until now used the remote API to test everything...

I noticed that changes in arming mode do come up as push notifications, so I will start by recording the mode. This also helps in creating an integration with Home Assistant, because after sending a command to change the mode, HA can show that it actually worked. I think I might be able to use the Alarm control panel integration (https://www.home-assistant.io/integrations/alarm_control_panel.mqtt/)... 

In order to prepare this, can some of you help identify the different push payloads. I think the message format would be based on the Home Base, of which there are at least two versions. To export the payloads of your Home Base:

1. Open a terminal and navigate to the "data" folder
2. Run `sqlite3 -csv database.sqlite "select * from push_payloads where type = 0"  > export.csv`
3. Open the file in an editor and remove personal information. Your HomeBase serial number will start with the model number, so please leave the first 6 characters of the serial number. For example `T8010PXXXXXXXXX`
4. Upload it here, together with your HomeBase type (HomeBase 1 or HomeBase 2 or ..?)

Example for Home Base 2:

```
308CF908|T8010PXXXXXXXXXX||0|{"id":"308CF908","from":"XXXXXXXXXXXXX","to":"XXXXXXXXXXXXXXXXXXX","category":"com.oceanwing.battery.cam","persistentId":"0:1611196396582151%0d2a775cf9fd7ecd","ttl":3600,"sent":"1611196396578","payload":{"content":"Arming change","device_sn":"","event_time":"1611196396101","payload":{"a":9,"s":"T8010PXXXXXXXXXX","t":"1611196396","arming":1,"mode":1,"alarm":0,"alarm_delay":0,"user":2,"user_name":"XXXXXXXXXXXXXXXXXX"},"push_time":"1611196396566","station_sn":"T8010PXXXXXXXXXX","title":"Eufy Security","type":"0","google.c.sender.id":"348804314802"}}
```

If you also have a Home Base 2, I only need all the data if it is completely different... if the structure is the same, it's enough to just mention that... Hopefully the structure is the same for everybody, which makes it easier to implement

**Comment by mikedrawback** [comment]
I have the floodlight camera, and indoor camera and the battery doorbell. 

The floodlight and indoor camera are currently not supported by the homebase. These cameras do not have a notification when guard mode changes, but maybe there is a way to poll them locally for the current mode. This wouldn't be a concern for batter drain since they are both AC powered.

The home base does send notifications about how the guard mode was changed; schedule, geofence, app, or keypad

**Comment by beck22179** [comment]
I have a Home Base 2 and can confirm the payload is the same.

**Comment by vlumikero** [comment]
I'm using the wired T8200, and it looks like the mode change events aren't logged in the DB. Do I need to do anything to enable that?

**Comment by matijse** [comment]
I assume you are using the doorbell without a Home Base? I think it needs to be polled for the state changes. 

I am working on switching to the new library that should be able to communicate directly with devices, but it is not going very fast, as I don't have a lot of time at the moment...

**Comment by vlumikero** [comment]
> I assume you are using the doorbell without a Home Base? I think it needs to be polled for the state changes.

Yes, that is correct. Well, do tell if there is anything you need support with. I should be able to do testing and limited debugging at least.

**Comment by r77s** [comment]
> Yes this one is next on my todo list :) It's a bit more tricky, because it looks like it is only possible to send this command over the local network directly to the Home Base. I don't actually own any Eufy camera's, I was just developing this for friends and up until now used the remote API to test everything...
> 
> I noticed that changes in arming mode do come up as push notifications, so I will start by recording the mode. This also helps in creating an integration with Home Assistant, because after sending a command to change the mode, HA can show that it actually worked. I think I might be able to use the Alarm control panel integration (https://www.home-assistant.io/integrations/alarm_control_panel.mqtt/)...
> 
> In order to prepare this, can some of you help identify the different push payloads. I think the message format would be based on the Home Base, of which there are at least two versions. To export the payloads of your Home Base:
> 
> 1. Open a terminal and navigate to the "data" folder
> 2. Run `sqlite3 -csv database.sqlite "select * from push_payloads where type = 0"  > export.csv`
> 3. Open the file in an editor and remove personal information. Your HomeBase serial number will start with the model number, so please leave the first 6 characters of the serial number. For example `T8010PXXXXXXXXX`
> 4. Upload it here, together with your HomeBase type (HomeBase 1 or HomeBase 2 or ..?)
> 
> Example for Home Base 2:
> 
> ```
> 308CF908|T8010PXXXXXXXXXX||0|{"id":"308CF908","from":"XXXXXXXXXXXXX","to":"XXXXXXXXXXXXXXXXXXX","category":"com.oceanwing.battery.cam","persistentId":"0:1611196396582151%0d2a775cf9fd7ecd","ttl":3600,"sent":"1611196396578","payload":{"content":"Arming change","device_sn":"","event_time":"1611196396101","payload":{"a":9,"s":"T8010PXXXXXXXXXX","t":"1611196396","arming":1,"mode":1,"alarm":0,"alarm_delay":0,"user":2,"user_name":"XXXXXXXXXXXXXXXXXX"},"push_time":"1611196396566","station_sn":"T8010PXXXXXXXXXX","title":"Eufy Security","type":"0","google.c.sender.id":"348804314802"}}
> ```
> 
> If you also have a Home Base 2, I only need all the data if it is completely different... if the structure is the same, it's enough to just mention that... Hopefully the structure is the same for everybody, which makes it easier to implement

@matijse: Do you still need the data? Sorry for my late responce. I own a eufycam 2c and can create the data if need. For me personal I have one use case:
* Create with HA an automation to change to geofencing from 9 to 18 o 'clock and in the other time to arm.
So I will one need the option to change the modi in the eufy entitie.

**Comment by r77s** [comment]
@matijse : Do you know this project?

[https://github.com/fuatakgun/eufy_security](url)

I didn't test it yet, but when I read the docs, the mode change with geo is possible.

**Comment by matijse** [comment]
@r77s Yes, I've used it for some time now and it works well... I can recommend that integration.

Since it already has more features, I decided to no longer actively develop this... Will update the readme to reflect this...
